In [1]:
import pandas as pd 

In [2]:
df = pd.read_csv(r"C:\Protfolio_Project\Project_5\Sample-superstore-data.csv", encoding="ISO-8859-1")

# Data Understanding

In [3]:
df.shape

(9994, 21)

In [4]:
df.info()

# Convert 'Ship Date' and 'Order Date' from string to datetime

<class 'pandas.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         9994 non-null   int64  
 1   Order ID       9994 non-null   str    
 2   Order Date     9994 non-null   str    
 3   Ship Date      9994 non-null   str    
 4   Ship Mode      9994 non-null   str    
 5   Customer ID    9994 non-null   str    
 6   Customer Name  9994 non-null   str    
 7   Segment        9994 non-null   str    
 8   Country        9994 non-null   str    
 9   City           9994 non-null   str    
 10  State          9994 non-null   str    
 11  Postal Code    9983 non-null   float64
 12  Region         9994 non-null   str    
 13  Product ID     9994 non-null   str    
 14  Category       9994 non-null   str    
 15  Sub-Category   9994 non-null   str    
 16  Product Name   9994 non-null   str    
 17  Sales          9994 non-null   float64
 18  Quantity       9994

In [5]:
df.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
0,1,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420.0,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600,2,0.00,41.9136
1,2,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420.0,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400,3,0.00,219.5820
2,3,CA-2016-138688,6/12/2016,6/16/2016,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,...,90036.0,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200,2,0.00,6.8714
3,4,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311.0,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.0310
4,5,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311.0,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680,2,0.20,2.5164


In [6]:
# Check the number of missing values in each column
df.isnull().sum()

Row ID            0
Order ID          0
Order Date        0
Ship Date         0
Ship Mode         0
Customer ID       0
Customer Name     0
Segment           0
Country           0
City              0
State             0
Postal Code      11
Region            0
Product ID        0
Category          0
Sub-Category      0
Product Name      0
Sales             0
Quantity          0
Discount          0
Profit            0
dtype: int64

In [7]:
# Check for duplicate rows in the dataset
df.duplicated().sum()

np.int64(0)

In [8]:
# Generate descriptive statistics for numerical columns
df.describe()

,Row ID,Postal Code,Sales,Quantity,Discount,Profit
count,9994.000000,9983.000000,9994.000000,9994.000000,9994.000000,9994.000000
mean,4997.500000,55245.233297,229.858001,3.789574,0.156203,28.656896
std,2885.163629,32038.715955,623.245101,2.225110,0.206452,234.260108
min,1.000000,1040.000000,0.444000,1.000000,0.000000,-6599.978000
25%,2499.250000,23223.000000,17.280000,2.000000,0.000000,1.728750
50%,4997.500000,57103.000000,54.490000,3.000000,0.200000,8.666500
75%,7495.750000,90008.000000,209.940000,5.000000,0.200000,29.364000
max,9994.000000,99301.000000,22638.480000,14.000000,0.800000,8399.976000


In [9]:
# Generate summary statistics for categorical columns
df.describe(include=['object'])

C:\Users\mayma\AppData\Local\Temp\ipykernel_27464\3390561020.py:2: Pandas4Warning: For backward compatibility, 'str' dtypes are included by select_dtypes when 'object' dtype is specified. This behavior is deprecated and will be removed in a future version. Explicitly pass 'str' to `include` to select them, or to `exclude` to remove them and silence this warning.
See https://pandas.pydata.org/docs/user_guide/migration-3-strings.html#string-migration-select-dtypes for details on how to write code that works with pandas 2 and 3.
  df.describe(include=['object'])


,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,State,Region,Product ID,Category,Sub-Category,Product Name
count,9994,9994,9994,9994,9994,9994,9994,9994,9994,9994,9994,9994,9994,9994,9994
unique,5009,1237,1334,4,793,793,3,1,531,49,4,1862,3,17,1850
top,CA-2017-100111,9/5/2016,12/16/2015,Standard Class,WB-21850,William Brown,Consumer,United States,New York City,California,West,OFF-PA-10001970,Office Supplies,Binders,Staple envelope
freq,14,38,35,5968,37,37,5191,9994,915,2001,3203,19,6026,1523,48


# Data Cleaning

In [10]:
# Convert date columns to datetime format for time-based analysis
df['Order Date'] =pd.to_datetime(df['Order Date'], format='%m/%d/%Y')
df['Ship Date'] = pd.to_datetime(df['Ship Date'], format='%m/%d/%Y')

df.info()

<class 'pandas.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   Row ID         9994 non-null   int64         
 1   Order ID       9994 non-null   str           
 2   Order Date     9994 non-null   datetime64[us]
 3   Ship Date      9994 non-null   datetime64[us]
 4   Ship Mode      9994 non-null   str           
 5   Customer ID    9994 non-null   str           
 6   Customer Name  9994 non-null   str           
 7   Segment        9994 non-null   str           
 8   Country        9994 non-null   str           
 9   City           9994 non-null   str           
 10  State          9994 non-null   str           
 11  Postal Code    9983 non-null   float64       
 12  Region         9994 non-null   str           
 13  Product ID     9994 non-null   str           
 14  Category       9994 non-null   str           
 15  Sub-Category   9994 non-null   s

# EDA

In [11]:
# Calculate total sales generated by the business
total_sales = df['Sales'].sum()

# Calculate total profit earned by the business
total_profit = df['Profit'].sum()

# Count the total number of orders
total_orders = df['Order ID'].nunique()

# Calculate the average sales per order
average_sales_per_order = total_sales / total_orders

total_sales, total_profit, total_orders, average_sales_per_order

(np.float64(2297200.8603),
 np.float64(286397.0217),
 5009,
 np.float64(458.6146656618087))

Total Sales = 2297200.86

Total Profit = 286397.02

Total Orders = 5009

Average Sales per Order = 458.61

## Sales Trend Analysis

In this section, we analyze sales performance over time by extracting year and month information from the order date. This enables us to identify growth trends, seasonality, and year-over-year performance.

In [12]:
# Extract the order year for yearly trend analysis
df['Order Year'] = df['Order Date'].dt.year

# Extract the month name for monthly reporting
df['Order Month Name'] = df['Order Date'].dt.month_name()

# Extract the month number to maintain chronological month sorting
df['Order Month Num'] = df['Order Date'].dt.month

## Yearly Sales and Profit Analysis

This analysis examines yearly sales and profit trends to evaluate business growth over time and identify changes in overall performance.

In [13]:
# Calculate yearly sales and profit to analyze business performance over time


yearly_performance  = (
    df.groupby('Order Year')[["Sales", "Profit"]]
    .sum()
    .reset_index()
)

yearly_performance

,Order Year,Sales,Profit
0,2014,484247.4981,49543.9741
1,2015,470532.5090,61618.6037
2,2016,609205.5980,81795.1743
3,2017,733215.2552,93439.2696


In [14]:
# Analyze average profit by discount level

discount_analysis = (
    df.groupby('Discount')[["Sales","Profit"]]
    .mean()
    .reset_index()
    .sort_values("Discount")
)
discount_analysis


,Discount,Sales,Profit
0,0.00,226.742074,66.900292
1,0.10,578.397351,96.055074
2,0.15,529.971567,27.288298
3,0.20,209.076940,24.702572
4,0.30,454.742974,-45.679636
5,0.32,536.794770,-88.560656
6,0.40,565.134874,-111.927429
7,0.45,498.634000,-226.646464
8,0.50,892.705152,-310.703456
9,0.60,48.150000,-43.077212


In [15]:
# Analyze sales and profit by product category
category_performance = (
    df.groupby('Category')[["Sales", "Profit"]]
    .sum()
    .sort_values("Sales", ascending=False)
)

category_performance

,Sales,Profit
Category,,
Technology,836154.0330,145454.9481
Furniture,741999.7953,18451.2728
Office Supplies,719047.0320,122490.8008


## Furniture Sub-Category Analysis

This analysis drills down into the Furniture category to identify which sub-categories contribute positively or negatively to overall profitability.

In [16]:
# Analyze sales and profit for Furniture sub-categories

furniture_performance = (
    df.groupby('Sub-Category')[['Sales', 'Profit']]
    .sum()
    .sort_values('Profit', ascending=False)
)

furniture_performance

,Sales,Profit
Sub-Category,,
Copiers,149528.0300,55617.8249
Phones,330007.0540,44515.7306
Accessories,167380.3180,41936.6357
Paper,78479.2060,34053.5693
Binders,203412.7330,30221.7633
Chairs,328449.1030,26590.1663
Storage,223843.6080,21278.8264
Appliances,107532.1610,18138.0054
Furnishings,91705.1640,13059.1436


## Furniture Profit by Region

This analysis evaluates the profitability of Furniture across different regions to determine whether losses are concentrated in specific geographic areas.

In [17]:
# Analyze sales and profit across regions for the Furniture category to identify geographic areas contributing to low profitability
furniture_analysis = (
    df[df['Category'] == 'Furniture']
    .groupby('Region')[['Sales', 'Profit']]
    .sum()
    .sort_values('Profit', ascending=False)
)

furniture_analysis


,Sales,Profit
Region,,
West,252612.7435,11504.9503
South,117298.6840,6771.2061
East,208291.2040,3046.1658
Central,163797.1638,-2871.0494


## Discount Analysis for Furniture in the Central Region

This analysis investigates whether high discounts are contributing to the negative profitability of Furniture products in the Central region.

In [18]:
# Filter the dataset to include only Furniture sales in the Central region

central_furniture_sales = df[(df['Category'] == 'Furniture') & (df['Region'] == 'Central')]

# Sort the results by profit to identify the most and least profitable discount levels
central_discount_analysis = (
    central_furniture_sales.groupby('Discount')[['Sales', 'Profit']]
    .sum()
    .sort_values('Profit', ascending=False)
)

central_discount_analysis

,Sales,Profit
Discount,,
0.00,74929.3500,16641.3819
0.32,14493.4588,-2391.1377
0.50,6550.6700,-4309.7447
0.60,6644.7000,-5944.6552
0.30,61178.9850,-6866.8937


## Product-Level Analysis for Furniture in the Central Region

This analysis identifies the least profitable Furniture products in the Central region to determine whether losses are concentrated in a small number of products or distributed across the entire category.

In [19]:
# Filter the dataset to include only Furniture sales in the Central region
central_furniture_sales = df[(df['Category'] == 'Furniture') & (df['Region'] == 'Central')]

# Analyze sales and profit by product to identify the least profitable products
central_furniture_product_analysis = (
    central_furniture_sales.groupby('Product Name')[['Sales', 'Profit']]
    .sum()
    .sort_values('Profit', ascending=True)
)

central_furniture_product_analysis

,Sales,Profit
Product Name,,
"Tenex Chairmat w/ Average Lip, 45"" x 53""",544.8960,-681.1200
"Bevis Oval Conference Table, Walnut",1565.8800,-600.2540
BoxOffice By Design Rectangular and Half-Moon Meeting Room Tables,765.6250,-566.5625
O'Sullivan Plantations 2-Door Library in Landvery Oak,2323.3288,-546.6656
Bretford “Just In Time” Height-Adjustable Multi-Task Work Tables,626.1000,-538.4460
...,...,...
Hon GuestStacker Chair,2266.7000,589.3420
Global Executive Mid-Back Manager's Chair,2327.8400,605.2384
Office Star - Professional Matrix Back Chair with 2-to-1 Synchro Tilt and Mesh Fabric Seat,2807.8400,673.8816


## Category Profit Margin Analysis

This analysis evaluates the profit margin for each product category to measure profitability relative to sales rather than relying on profit alone.

In [26]:
# Calculate total sales and profit for each product category
total_sales_profit_category = (
    df.groupby('Category')[['Sales', 'Profit']]
    .sum()
    .sort_values('Profit', ascending=False)
    .reset_index()
)
total_sales_profit_category

,Category,Sales,Profit
0,Technology,836154.0330,145454.9481
1,Office Supplies,719047.0320,122490.8008
2,Furniture,741999.7953,18451.2728


In [30]:
# Calculate the profit margin for each product category
total_sales_profit_category['Profit Margin'] = (
    (total_sales_profit_category['Profit'] / total_sales_profit_category['Sales']) * 100

)
total_sales_profit_category[['Category', 'Profit Margin']]

,Category,Profit Margin
0,Technology,17.395712
1,Office Supplies,17.035158
2,Furniture,2.486695


In [31]:
# Sort categories by profit margin to compare overall profitability
total_sales_profit_category.sort_values('Profit Margin', ascending=False)

,Category,Sales,Profit,Profit Margin
0,Technology,836154.0330,145454.9481,17.395712
1,Office Supplies,719047.0320,122490.8008,17.035158
2,Furniture,741999.7953,18451.2728,2.486695


## Shipping Mode Analysis for Furniture in the Central Region

This analysis investigates whether specific shipping modes are associated with lower profitability for Furniture sales in the Central region.

In [33]:
# Filter the dataset to include only Furniture sales in the Central region
furniture_central_region = df[(df['Category'] == 'Furniture') & (df['Region'] == 'Central')]

In [35]:
# Analyze sales and profit by shipping mode
furniture_central_region_shipping_analysis = (
    furniture_central_region.groupby('Ship Mode')[['Sales', 'Profit']]
    .sum()
    .reset_index()
)
furniture_central_region_shipping_analysis

,Ship Mode,Sales,Profit
0,First Class,21037.8254,-660.8860
1,Same Day,8842.2860,-207.1337
2,Second Class,31187.1234,-1580.8283
3,Standard Class,102729.9290,-422.2014


In [37]:
# Sort shipping modes by profit to identify the least profitable delivery methods
furniture_central_region_shipping_analysis.sort_values('Profit', ascending=False)

,Ship Mode,Sales,Profit
1,Same Day,8842.2860,-207.1337
3,Standard Class,102729.9290,-422.2014
0,First Class,21037.8254,-660.8860
2,Second Class,31187.1234,-1580.8283


## Shipping Mode Performance Analysis

This analysis compares shipping modes using both total and per-order metrics to determine whether profitability differences are driven by shipping performance or simply by order volume.

In [40]:
# Analyze order volume, sales, and profit by shipping mode
furniture_central_region_order_analysis = (
    furniture_central_region.groupby('Ship Mode').agg(
        Total_Sales=('Sales', 'sum'),
        Total_Profit=('Profit', 'sum'),
        Order_Count=('Order ID', 'nunique'),
    )
)

furniture_central_region_order_analysis

,Total_Sales,Total_Profit,Order_Count
Ship Mode,,,
First Class,21037.8254,-660.8860,59
Same Day,8842.2860,-207.1337,21
Second Class,31187.1234,-1580.8283,76
Standard Class,102729.9290,-422.2014,247


In [44]:
# Calculate average sales and profit per order for each shipping mode
furniture_central_region_order_analysis['average_sales_per_order'] = furniture_central_region_order_analysis['Total_Sales'] / furniture_central_region_order_analysis['Order_Count']
furniture_central_region_order_analysis['average_profit_per_order'] = furniture_central_region_order_analysis['Total_Profit'] / furniture_central_region_order_analysis['Order_Count'] 

furniture_central_region_order_analysis.sort_values('average_profit_per_order', ascending=False)

,Total_Sales,Total_Profit,Order_Count,average_sales_per_order,average_profit_per_order
Ship Mode,,,,,
Standard Class,102729.9290,-422.2014,247,415.910644,-1.709317
Same Day,8842.2860,-207.1337,21,421.061238,-9.863510
First Class,21037.8254,-660.8860,59,356.573312,-11.201458
Second Class,31187.1234,-1580.8283,76,410.356887,-20.800372
